modules/vpn_aws_gcp/aws.tf

```diff
@@ -0,0 +1,46 @@
+resource "aws_vpn_gateway" "a" {
+  count = var.use_existing_aws_vgw ? 0 : 1
+
+  # A VPC can only have one VPN gateway attached
+  vpc_id          = var.aws_vpc_id
+  amazon_side_asn = var.aws_bgp_asn
+}
+
+resource "aws_vpn_gateway_route_propagation" "a" {
+  count = length(data.aws_route_tables.tables.ids)
+
+  vpn_gateway_id = var.use_existing_aws_vgw ? data.aws_vpn_gateway.existing[0].id : aws_vpn_gateway.a[0].id
```

Comment:
`var.use_existing_aws_vgw ? data.aws_vpn_gateway.existing[0].id : aws_vpn_gateway.a[0].id` can be set as a local variable to avoid having to specify the same code again multiple times.

Reply:
good point